Pidaq/Lib/uploadToDrive/uploadToDrive.py

```diff
@@ -0,0 +1,125 @@
+
+
+SCOPES = 'https://www.googleapis.com/auth/drive'
+store = file.Storage('credentials.json') 
+creds = store.get()
+
+if not creds or creds.invalid:
+    flow = client.flow_from_clientsecrets('client_secret.json', SCOPES)
+    creds = tools.run_flow(flow, store)
+    
+DRIVE = build('drive', 'v3', http=creds.authorize(Http()))
+
+# Searches for the Json file containing folder names and IDS
+# These IDs are to be stored manually.
+try:
+    with open('dictionary.json') as json_file:    
+        parentDictionary = json.load(json_file)
+except:
+    logging.error('Parents dictionary .json file not found')
+    exit()
+# change this for the ID of the default folder 
+DefaultID = '1VDOOjuOeyNRJgdKmmRTJC1HVjq91F4Q8'
+
+def uploadCsv(filePath, parentName):
+    """ Uploads a csv file from a local directory to a folder on Google Drive.
+
+    Parameters:
+    "filePath" is the path of the csv file itself and
+                should be passed without quotations.
+    "parentName" is the name of the testing folder in the drive
+            and only a specific set of values are allowed.
+    The parent folder is supposedly going to be fixed for a
+    long period of time.
+    """
+    # searches for the parent name in the dictionary
+    for key in parentDictionary.keys():
+        if key.lower() == parentName.lower():
+            parentKey = key
+    try:
+        parentFolderID = parentDictionary[parentKey]
+    except KeyError:
+        # uploads to a default folder if the folder name does
+        # not exist in the dictionary
+        parentFolderID = DefaultID
+        logging.error('Folder name not found \nuploading to "Default"')
+    
+    #Today's date in YYYY-MM-DD format is stored in the variable "date"
+    date = datetime.today().strftime('%Y-%m-%d')
+    
+    # If the folder does not exist, should be the case
+    # when running for the first time on a given day
+    if checkExistingSubFolder(date, parentFolderID) == 0:
+        folder_metadata = {
+            'name' : date,
+            'mimeType': 'application/vnd.google-apps.folder',
+            'parents' : [parentFolderID]
+            }
+        # Creates a folder with the given date
+        folder = DRIVE.files().create(body = folder_metadata,
+                                      fields = 'id').execute()
+        # Returns the date folder's ID
+        folderId = getFolderId(date, parentFolderID)
+    # executes if the folder already exists  
+    elif checkExistingSubFolder(date, parentFolderID) == 1:
+        logging.warning("found folder for today's date")
+        folderId = getFolderId(date, parentFolderID)
+    # executes if more than one folder exists
+    else:
+        logging.warning('unexpected count of folders named with a given date')
+        return
+    
+    # extracts the .csv file name from the path
+    fileName=os.path.basename(filePath)
+    metadata = {'name': fileName,
+                'parents' : [folderId]
+                }

```

Comment:
The way you formatted the dict on lines 62-66 looked a bit neater

Reply:
I suggest making this dict in one line. The other one has 3 items and is longer. This one originally had 3 before we decided to remove the item that caused an error on one of the laptops.